bin/build

```diff
@@ -7,4 +7,4 @@ echo -e "\n Formatting packages..."
   go fmt ./...
 
 echo -e "\nGenerating Binary..."
-go build -o $(dirname $0)/../out/slgo ./main/slgo.go
+go build -o $(dirname $0)/../out/slgo ../main/slgo.go
```

Comment:
Cool thanks. I will complete my review soon. I think overall we might be close. 

One thing I want to do first before merging is getting one of the projects we have using softlayer-go testing with your branch first. Sort of `develop` branch though we have not officially moved this this develop/master process.

More soon.